File: accsaber-database/src/main/java/de/ixsen/accsaber/database/views/extenders/WithRankAndStats.java
package de.ixsen.accsaber.database.views.extenders;

import org.hibernate.annotations.Immutable;

import javax.persistence.MappedSuperclass;

@MappedSuperclass
@Immutable
public abstract class WithRankAndStats extends WithRank {

    private Double averageAcc;
    private Double averageApPerMap;
    private int rankedPlays;

    public Double getAverageApPerMap() {
        return this.averageApPerMap;
    }

    public int getRankedPlays() {
        return this.rankedPlays;
    }

    public Double getAverageAcc() {
        return this.averageAcc;
    }
}

File: accsaber-database/src/main/kotlin/de/ixsen/accsaber/database/views/extenders/WithRankAndStats.kt
package de.ixsen.accsaber.database.views.extenders

import org.hibernate.annotations.Immutable
import javax.persistence.MappedSuperclass

@MappedSuperclass
@Immutable
abstract class WithRankAndStats : WithRank() {
    val averageAcc: Double? = null
    val averageApPerMap: Double? = null
    val rankedPlays = 0
}